woocommerce/woocommerce
# Refactor product tabs and add product tab slot fills

### All Submissions:






### Changes proposed in this Pull Request:



Closes #36446 and #36394  .






### How to test the changes in this Pull Request:



1. Checkout branch.
2. Enable product editor at WooCommerce -> Settings -> Advanced -> Features -> Try the new product editor
3. Go to product editor at Products -> Add new (MVP)
4. Do an overall smoke test of the tabs and sections within each tab
5. Make sure to enable the `product-variation-management` feature flag and add some options and click `edit` on one of the option/variations, as this uses a different set of tabs.
6. You may also want to test the slot fill by adding this code somewhere in the app or in your mu-plugin:
```
import {
	__experimentalWooProductTabItem as WooProductTabItem,
} from '@woocommerce/components';

function Tabs() {
  return <WooProductTabItem
				id={ 'tab/new-tab' }
				location="tab/general"
				pluginId="test-plugin"
				order={ 3 }
				tabProps={ { name: 'new-tab', title: 'New Tab' } }
			>
	<div>Hi this is the content of this new slot filled tab.</div>
  </WooProductTabItem>
}

registerPlugin( 'wc-admin-product-editor-form-extra-tab-fills', {
	scope: 'woocommerce-product-editor',
	render: () => {
		return <Tabs />;
	},
} );
```



### Other information:




### FOR PR REVIEWER ONLY:
